Source/SwiftyJSON.swift

```diff
@@ -428,7 +428,7 @@ extension JSON {
         }
         set {
             if self.type == .array {
-                if self.rawArray.count > index && newValue.error == nil {
+                if index >= 0 && index < self.rawArray.count && newValue.error == nil {
```

Comment:
https://github.com/SwiftyJSON/SwiftyJSON/pull/852/files#diff-b89f704655f1cdfa5eef8141e87c1abaR421 should be refactored too.